db/scripts/2_create_l1_tables.sql

```diff
@@ -0,0 +1,106 @@
+
+\connect rollup;
+
+CREATE TABLE l1_block (
+  id BIGSERIAL NOT NULL,
+  block_hash CHARACTER(66) NOT NULL,
+  parent_hash CHARACTER(66) NOT NULL,
+  block_number BIGINT NOT NULL,
+  block_timestamp BIGINT NOT NULL,
+  gas_limit NUMERIC(78) NOT NULL,
+  gas_used NUMERIC(78) NOT NULL,
+  processed BOOLEAN NOT NULL DEFAULT FALSE,
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (block_hash),
+  UNIQUE (block_number)
+);
+
+
+CREATE TABLE l1_tx (
+  id BIGSERIAL NOT NULL,
+  block_number BIGINT NOT NULL,
+  tx_hash CHARACTER(66) NOT NULL,
+  from_address CHARACTER(42) NOT NULL,
+  to_address CHARACTER(42) NOT NULL,
+  nonce NUMERIC(78) NOT NULL,
+  gas_limit NUMERIC(78) NOT NULL,
+  gas_price NUMERIC(78) NOT NULL,
+  calldata TEXT NOT NULL,
+  signature CHARACTER(130) NOT NULL,
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (tx_hash),
+  FOREIGN KEY (block_number) REFERENCES l1_block(block_number)
+);
+
+CREATE TABLE geth_submission_queue (
+  id BIGSERIAL NOT NULL,
+  l1_tx_hash CHARACTER(66) NOT NULL,
+  queue_index BIGINT NOT NULL,
+  status TEXT NOT NULL DEFAULT 'QUEUED',
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (queue_index),
+  FOREIGN KEY (l1_tx_hash) REFERENCES l1_tx (tx_hash)
+);
+CREATE INDEX geth_submission_queue_status_idx ON geth_submission_queue USING btree (status);
+
+
+CREATE TABLE l1_state_root_batch (
+  id BIGSERIAL NOT NULL,
+  l1_tx_hash CHARACTER(66) NOT NULL,
+  batch_number BIGINT NOT NULL,
+  status TEXT NOT NULL DEFAULT 'UNVERIFIED',
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (batch_number),
+  FOREIGN KEY (l1_tx_hash) REFERENCES l1_tx (tx_hash)
+);
+CREATE INDEX l1_state_root_batch_status_idx ON l1_state_root_batch USING btree (status);
+
+
+
+CREATE TABLE l1_state_root (
+  id BIGSERIAL NOT NULL,
+  state_root character(66) NOT NULL,
+  batch_number BIGINT NOT NULL,
+  batch_index INT NOT NULL,
+  removed BOOLEAN NOT NULL DEFAULT FALSE,
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (state_root),
+  FOREIGN KEY (batch_number) REFERENCES l1_state_root_batch (batch_number)
+);
+
+
+
+CREATE TABLE l1_rollup_tx (
+  id BIGSERIAL NOT NULL,
+  sender CHARACTER(42) DEFAULT NULL,
+  l1_message_sender CHARACTER(42) DEFAULT NULL,
+  target CHARACTER(42) DEFAULT NULL,
+  calldata TEXT DEFAULT NULL,
+  queue_origin SMALLINT NOT NULL,
+  nonce NUMERIC(78) DEFAULT NULL,
+  gas_limit NUMERIC(78) DEFAULT NULL,
+  signature CHARACTER(132) DEFAULT NULL,
+  geth_submission_queue_index BIGINT DEFAULT NULL,
+  index_within_submission INT DEFAULT NULL,
```

Comment:
A little confused by this val--is this index within "geth submission",  or something else?

Reply:
Yep. Geth submissions are an ordered list of `l1_rollup_tx`. This is how they're ordered.